### PR TITLE
Wp

### PR DESCRIPTION
**Pull request by PotosnakW**
Hi Chirag, 

Here is a first draft of the phenotyping notebook titled, "Demo of Phenotyping on Support Data". I'd really appreciate your feedback when you can get a chance to review it. I'll start on the "estimators" notebook in the meantime. 

I also made a couple small edits to phenotyping.py and preprocessing.py.

### CODE REVIEW

File: auton_survival/phenotyping.py [hunk 1]
```diff
@@ -135,16 +135,18 @@ def phenotype(self, features):
...
     assert self.fitted, "Phenotyper must be `fitted` before calling `phenotype`."
     features = deepcopy(features)
 
     for num_var in self.num_vars:
 
       var_min, var_max = self.min_max[num_var]
-
-      features[num_var][features[num_var]>=var_max] = var_max
-      features[num_var][features[num_var]<=var_min] = var_min
+
```
Comment (chiragnagpal) [inline auton_survival/phenotyping.py:147]: lets leave this as is, the original approach is somewhat more readabile (to me atleast)

File: auton_survival/phenotyping.py [hunk 2]
```diff
@@ -180,6 +182,7 @@ def _rename(self, phenotypes):
         for j in range(len(ft_names)):
             row.append(ft_names[j]+":"+str(phenotypes[i][j]))
         renamed.append(" & ".join(row))
+
```
Comment (chiragnagpal) [inline auton_survival/phenotyping.py:185]: remove extra line

File: auton_survival/preprocessing.py [hunk 3]
```diff
@@ -178,9 +178,61 @@ def __init__(self, scaling_strategy='standard'):
...
+    if self.scaling_strategy == 'standard':
+      scaler = StandardScaler()
+    elif self.scaling_strategy == 'minmax':
+      scaler = MinMaxScaler()
+    else:
+      scaler = None
+
+    if scaler is not None:
+      if feats: self.scaler = scaler.fit(df[feats])
+      else: self.scaler = scaler.fit_(df)
```
Comment (chiragnagpal) [inline auton_survival/preprocessing.py:208]: why `fit_` ?

File: auton_survival/preprocessing.py [hunk 4]
```diff
@@ -178,9 +178,61 @@ def __init__(self, scaling_strategy='standard'):
...
+        
+    self.fitted = True
+    return self
+
+  def transform(self, data, feats=[]):
```
Comment (chiragnagpal) [inline auton_survival/preprocessing.py:213]: store the feats in the `fit` function

File: auton_survival/preprocessing.py [hunk 5]
```diff
@@ -178,9 +178,61 @@ def __init__(self, scaling_strategy='standard'):
...
+    
+    df = data.copy()
+
+    if self.scaling_strategy == 'standard':
+      scaler = StandardScaler()
+    elif self.scaling_strategy == 'minmax':
+      scaler = MinMaxScaler()
+    else:
+      scaler = None
+
```
Comment (chiragnagpal) [inline auton_survival/preprocessing.py:205]: store the passed feats `feats` in the fit function

File: auton_survival/preprocessing.py (hunk 4 above)
Comment (chiragnagpal) [reply auton_survival/preprocessing.py:213]: transform should pick up the feats using the ones stored in the `fit` function

File: auton_survival/preprocessing.py [hunk 6]
```diff
@@ -178,9 +178,61 @@ def __init__(self, scaling_strategy='standard'):
...
+    df = data.copy()
+    
+    if feats: df[feats] = self.scaler.transform(df[feats])
+    else: df[df.columns] = self.scaler.transform(df)
+
+    return df
 
   def fit_transform(self, data, feats=[]):
-    """Scales dataset using the scaling strategy.
+    """Fits scaler and scales dataset using the scaling strategy.
```
Comment (chiragnagpal) [inline auton_survival/preprocessing.py:235]: Fits a scaler and rescales a dataset using a standard rescaling strategy

File: auton_survival/phenotyping.py (hunk 1 above)
Comment (PotosnakW) [reply auton_survival/phenotyping.py:147]: The original approach returns a warning in the notebook. The updated approach is recommended to do the same thing as the old approach without throwing a warning. Given this info, lmk if you'd still like me to change it back and I'll do so.

**Pull request closed without merging**